src/rust/engine/fs/store/src/snapshot_ops.rs

```diff
@@ -406,6 +406,12 @@ impl IntermediateGlobbedFilesAndDirectories {
           RestrictedPathGlob::Wildcard { wildcard } => {
             // NB: We interpret globs such that the *only* way to have a glob match the contents of
             // a whole directory is to end in '/**' or '/**/*'.
+
+            if exclude.maybe_is_parent_of_ignored_path(&directory_path) {
+              // leave this directory in todo_directories, so we process ignore patterns correctly..
+              continue;
```

Comment:
I'm not really sure about this one... but it did make my test pass. See next comment.